application/src/main/java/org/openhds/mobile/activity/NavigateActivity.java

```diff
@@ -322,12 +340,48 @@ private void hierarchySetup() {
             showDetailFragment();
             detailToggleFragment.setButtonHighlighted(true);
         }
+        //populate recent forms
+        populateFormsForPath();
 
         if(null != getCurrentVisit()){
             visitFragment.setButtonEnabled(true);
         }
     }
 
+    private void populateFormsForPath() {
+
+        List<FormInstance> recforms;
+        ContentResolver reslovr = this.getContentResolver();
+        DatabaseAdapter dbAdpt = new DatabaseAdapter(this);
+
+        //get all the existing recent form paths
+        Collection<String> formsdel=  dbAdpt.findAllFormPaths();
+
+        //find if any of the form is submitted
+        List <String> submitForms  = OdkCollectHelper.getSentFormPaths(getContentResolver(), formsdel);
```

Comment:
It seems strange to have the populate method delete the sent forms, especially from `NavigateActivity`. A few things:

  * Method contents should match the name of the method (I shouldn't be surprised; populating isn't deleting)
  * Responsibility should make sense for the class, fetching forms isn't, but deleting them from here is.

I probably would factor the deletion part out into its own method. I probably would move the code to a place that made more sense.